public/js/app.js

```diff
@@ -1,8 +1,86 @@
+/* eslint-disable no-unused-vars */
 const menu = document.querySelectorAll('.item-menu');
 const h1 = document.querySelector('h1');
 const contain = document.querySelector('.contain');
 const currentDate = document.querySelector('.current-date p');
-let books = [];
+
+class ListOfBooks {
+  books;
+
+  constructor() {
+    this.getData();
+  }
+
+  getData = () => {
+    const data = localStorage.getItem('datas');
+    if (data) {
+      this.books = JSON.parse(data);
+    } else {
+      this.books = [];
+    }
+  };
+
+  setLocalStorage = (newBooks) => localStorage.setItem('datas', JSON.stringify(newBooks));
+
+  addBook = (newBook) => {
+    this.books.push(newBook);
+    this.setLocalStorage(this.books);
+  };
+
+  removeBook = (index) => {
+    const filtered = this.books.filter(
+      (book) => book.title !== this.books[index].title,
+    );
+    localStorage.setItem('datas', JSON.stringify(filtered));
+  };
+
+  generateTable = () => {
+    contain.innerHTML = '';
+    const table = document.createElement('table');
+    // header
+    const tr = document.createElement('tr');
+    const th1 = document.createElement('th');
+    th1.textContent = '#';
+    const th2 = document.createElement('th');
+    th2.textContent = 'Title';
+    const th3 = document.createElement('th');
+    th3.textContent = 'Author';
+    const th4 = document.createElement('th');
+    th4.textContent = 'Created';
+    const th5 = document.createElement('th');
+    th5.textContent = 'Actions';
+    tr.appendChild(th1);
+    tr.appendChild(th2);
+    tr.appendChild(th3);
+    tr.appendChild(th4);
+    tr.appendChild(th5);
+    table.appendChild(tr);
+    // body
+    this.books.forEach((bk, i) => {
+      const tr1 = document.createElement('tr');
+      const th11 = document.createElement('th');
+      th11.textContent = i + 1;
+      const th12 = document.createElement('td');
+      th12.textContent = bk.title;
+      const th13 = document.createElement('td');
+      th13.textContent = bk.author;
+      const th14 = document.createElement('td');
+      th14.textContent = bk.created_at;
+      const th15 = document.createElement('td');
+      th15.innerHTML = `
+      <a href="#" onclick="popupBook(${i})"><i class="fa fa-search-plus" aria-hidden="true"></i></a> | 
+      <a href="#" onclick="removeBook(${i})"><i class="fa fa-trash-o" aria-hidden="true"></i></a>`;
+      tr1.appendChild(th11);
+      tr1.appendChild(th12);
+      tr1.appendChild(th13);
+      tr1.appendChild(th14);
+      tr1.appendChild(th15);
+      table.appendChild(tr1);
+    });
+    contain.appendChild(table);
+  };
+}
+const listofbooks = new ListOfBooks();
 
 /* eslint-disable no-unused-vars */
```

Comment:
- [ ] All so this part, I would strongly recommend you fix it up and remove the disabling lines as this will contribute to your professionalism and gain some good ways of writing a clean code in the near future, what do you think about it? 😃